journal/src/main/java/io/zeebe/journal/file/MappedJournalSegmentReader.java

```diff
@@ -119,6 +120,12 @@ long getNextIndex() {
 
   /** Reads the next entry in the segment. */
   private void readNext(final long expectedIndex) {
+    final Optional<Integer> version = FrameUtil.readVersion(buffer);
```

Comment:
Why not push this logic into reader util? This is a common logic used by both reader and writer when reading records. Not a blocker, just my opinion.

Reply:
Thought about it but decided against putting more stuff into the journal record reader that isn't reading a record. The class API was also small and simple to use directly, so seemed fine to leave it in the journal reader/writer